### PR TITLE
Add support for running a CMR deployment via the --bundle flag

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
When running a cross-model Zaza test, it is necessary to
be able to specify two (or more) different bundles in the
same command. This change adds support for multiple bundles,
along with their associated model aliases, to be passed in
and run correctly. This brings the --bundle argument inline
with the --dev, --gate, and --smoke deployment methods.

### CODE REVIEW

File: zaza/charm_lifecycle/func_test_runner.py [hunk 1]
```diff
@@ -232,9 +238,10 @@ def parse_args(args):
     parser.add_argument('--dev', dest='dev',
                         help='Just run dev test(s)',
                         action='store_true')
-    parser.add_argument('-b', '--bundle', dest='bundle',
+    parser.add_argument('-b', '--bundle', dest='bundles',
                         help='Override the bundle to be run',
```
Comment (AurelienLourot) [inline zaza/charm_lifecycle/func_test_runner.py:242]: Should we extend this description a bit?

File: zaza/charm_lifecycle/func_test_runner.py [hunk 2]
```diff
@@ -177,25 +177,31 @@ def func_test_runner(keep_model=False, smoke=False, dev=False, bundle=None,
     :param dev: Whether to just run dev test.
     :type smoke: boolean
     :type dev: boolean
+    :param bundles:
```
Comment (AurelienLourot) [inline zaza/charm_lifecycle/func_test_runner.py:180]: Should we describe this parameter?

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
LGTM in general, thanks!

File: zaza/charm_lifecycle/func_test_runner.py (hunk 1 above)
Comment (ChrisMacNaughton) [reply zaza/charm_lifecycle/func_test_runner.py:242]: Incidentally, changing the type does include usage directions that include multiple bundles:

```
usage: func_test_runner.py [-h] [--keep-model] [--smoke] [--dev] [-b BUNDLES [BUNDLES ...]] [-f] [--log LOGLEVEL] [--test-directory TEST_DIRECTORY]

optional arguments:
  -h, --help            show this help message and exit
  --keep-model          Keep model at the end of the run
  --smoke               Just run smoke test(s)
  --dev                 Just run dev test(s)
  -b BUNDLES [BUNDLES ...], --bundle BUNDLES [BUNDLES ...]
                        Override the bundle to be run
  -f, --force           Pass --force to the juju deploy command
  --log LOGLEVEL        Loglevel [DEBUG|INFO|WARN|ERROR|CRITICAL]
  --test-directory TEST_DIRECTORY
                        Set the directory containing the test config (test.yaml etc).
```

I'll update it to say "Override the bundles(s) to be run" as I do think that would be more clear.

**Review by AurelienLourot — APPROVED** [review]

**Pull request merged**